src/Network/GRPC/MQTT/RemoteClient.hs

```diff
@@ -238,11 +241,20 @@ handleRequest handle = do
     Left err -> do
       Session.logError "wire parse error encountered parsing Request" (show err)
       publishRemoteError Serial.defaultEncodeOptions (Message.toRemoteError err)
-    Right rqt@Request.Request{..} -> do
+    Right Request.Request{..} -> do
       let encodeOptions = Serial.makeRemoteEncodeOptions options
       let decodeOptions = Serial.makeClientDecodeOptions options
 
-      Session.logDebug "handling client request" (show rqt)
+      method <- askMethodTopic
+      Session.logInfo
+        "Received request"
+        ( Text.intercalate
+            ", "
+            [ "Method: " <> unTopic method
+            , "Metadata: " <> show metadata
```

Comment:
This will log tokens sent in the metadata header, right?

Reply:
Yes, it would. So I've gone and made the formatting of requests/responses configurable. Now you can pass in a formatter that redacts whatever fields you want.